source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/teamexplorer/internal/TeamExplorerHelpers.java

```diff
@@ -279,4 +311,219 @@ public static IViewPart showView(final String viewID) {
 
         return view;
     }
+
+    public static boolean tryParseProtocolHandlerArguments(final String[] applicationArgs) {
+
+        if (applicationArgs == null) {
+            return false;
+        }
+
+        boolean found = false;
+
+        for (final String arg : applicationArgs) {
+            if (found) {
+                final URI uri = URIUtils.newURI(arg);
```

Comment:
What does URIUtils.newURI return if the arg is not a valid URI? will this method throw or will uri end up null? You may need some kind of check here.